geps/gep-2659.md

```diff
@@ -0,0 +1,199 @@
+We will introduce the concept of GEP types.
+
+#### Types of GEP
+There will be two types of GEP. The existing type will be called `API` GEPs, and
+a new type `Memorandum` (after "Memorandum of Understanding") will be created.
+
+##### API GEPs
+
+GEPs that do not have an assigned type will be considered `API` GEPs. (For
+legacy GEPs). New GEPs created after this GEP is merged will default to the `API`
+type.
+
+As they do currently, these GEPs lay out a problem the API needs to address,
+along with a rationale and a solution for that problem.
+
+This also uses the existing GEP process and phases:
+
+1. Discuss
+1. Provisional
+  1. (optional) Prototyping
+1. Implementable
+1. Experimental
+1. Standard
+1. Completed
+
+As part of the addition of relationships, we will extend these phases, but see
+the "Addition of GEP relationships" section for details, but in short, when an
+API GEP has been Obsoleted, it will be moved to a new "Obsolete" phase (which
+is functionally the same as "Declined").
+
+##### Memorandum GEPs
+
+These GEPs record an agreement on either the definition of a problem, its scope
+and a common language, or further process changes to the GEP process itself.
+The defining characteristic here is that the GEP MUST NOT result in any changes
+to the Gateway API spec, and SHOULD result in further GEPs to further clarify.
+For GEPs that _do_ make changes to the API, but also require further GEPs to
+clarify, they SHOULD use the new "Updated By" relationship instead.
+
+Memorandum GEPs should be used sparingly, and should form the umbrella for a
+significant amount of work, particularly work that may have parts that can
+move through the GEP phases at different speeds.
+
+The process for Memorandum GEPs is much more simple, and has really only one
+main phase - "Accepted".
+
+This phase is reached when a Memorandum GEP is merged, although as we will document
+in the "Addition of GEP relationships" section, it can still be Updated
+or Obsoleted.
+
+Existing GEPs that meet this criteria will be gradually moved to be proper
+Memorandum GEPs after this GEP is merged.
+
+### Addition of YAML metadata file
+
+The core Gateway API mainatainers were hoping not to need metadata YAMLs for a
+while, but the addition of relationships has turbocharged the need for machine
+parseable GEP metadata.
+
+This should also help with building display for GEPs; theoretically we can build
+tooling that will let people slice the list of GEPs by whatever dimensions
+we (or they) wish.
+
+Similarly to the `ConformanceReport` object, I'm proposing to make CRD definitions
+for this `GEPDetails` object, which will not be included in the usual CRD
+definitions for Gateway API, nor will it be part of the regular API spec.
+
+The use of a CRD is just to make the schema more similar to the other schemas
+in this repository.
+
+With that said, here's a rough sample of what a GEPMetadata object will look like:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha1
```

Comment:
I'm not sure we need an APIVersion here, but if we have one, I think it should clearly denote that this is both internal only and backwards incompatible changes are expected without a version rev. I don't think it really makes sense for us to maintain multiple versions of these structs since they're intended purely for internal tooling. I also have similar thoughts about conformance reports, so hopefully whatever we do here could translate there as well.